node/lib/util/write_repo_ast_util.js

```diff
@@ -399,6 +400,24 @@ git checkout-index -a -f
         }
     }
 
+    // set up a rebase if there is one
+
+    if (null !== ast.rebase) {
+        const rebase = ast.rebase;
+        const originalSha = commitMap[rebase.originalHead];
+        const ontoSha = commitMap[rebase.onto];
+        const original = yield NodeGit.AnnotatedCommit.lookup(repo,
+                                                              originalSha);
+        const onto = yield NodeGit.AnnotatedCommit.lookup(repo, ontoSha);
+        yield NodeGit.Rebase.init(repo, original, onto, null, null);
```

Comment:
Something seems weird here -- the nodegit docs say that the args to Rebase.init are (repo, branch, upstream, onto, opts).